src/App.js

```diff
@@ -1,16 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages)
+
+  const onLike = id => {
+    setMessages((oldMessages) => {
+      return oldMessages.map((message) => {
+        if (id === message.id) {
+          return {...message, liked: !message.liked};
+        } else {
+          return message;
+        };
+      })
+    })
+  }
+
+  const calcTotalLikes = (messages) => {
+    let likedTally = 0;
+    for (let message of messages) {
+      if (message.liked) likedTally++; 
+    };
+    return likedTally
+  }
```

Comment:
Great use of the chat data to calculate the total hearts! Another way we could do this is with the Array.reduce function:
```js
let initialValue = 0;
return messages.reduce((total, message) => {
    return (chat.liked === true) ? total + 1 : total;
}, initialValue);
```